### PR TITLE
Profile/aw/54395/update profile dsot facilities selectors

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary
Main work:
- There was one last place that the old facilities list was being used within the authenticate experience applications and that was the 'Notification settings' page `profile/notifications`. This PR adds an application wide request to the Drupal Source Of Truth, so that the data is available in the Redux state store, no matter what page in Profile the user is viewing.
- In addition to requesting the data, the Notification Settings page was updated to use that state data by updating the selector that is being used.

Other 'bits and bobs':
- Updated the mock server to be slightly more organized with code being moved into an endpoints folder and a new script folder being added. For the most part all the changes in the `/mocks` directory are just a result of moving files into folders.

- New script `mockLocalDSOT.js` was created to make sure that when a mock server is started, that the static JSON asset file is present in the local build location so that it can be requested successfully. This new script is not used on the vets-website FE at all, but is just a local tool for running the mock server.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/54395

## Testing done

- Updated tests for Notification Settings
- Added unit tests for `AddContactInfoLink`, `ContactInfoOnFile`, and `MissingContactInfoAlert`

## Screenshots
No UI updates, just updated data that is being accessed and used.

## What areas of the site does it impact?
Profile - Notification Settings

## Acceptance criteria

- [x]  DSOT file used instead of static facility list.
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.

### CODE REVIEW

File: src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx [hunk 1]
```diff
@@ -56,6 +57,7 @@ const NotificationSettings = ({
 
     focusElement('[data-focus-target]');
     document.title = `Notification Settings | Veterans Affairs`;
+    // connectDrupalSourceOfTruthCerner(dispatch);
```
Comment (allisonlu) [inline src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx:60]: can you delete this?

**Comment by allisonlu** [comment]
ran e2e tests locally and got some errors, specifically for the following file:

**src/applications/personalization/profile/tests/e2e/profile-a11y.cypress.spec.js**
Profile -- should pass an aXe scan and manage focus on all pages at desktop size (failed)
Profile -- should pass an aXe scan and manage focus on all pages at mobile phone size (failed)

everything else looks good though 👍

File: src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx (hunk 1 above)
Comment (adamwhitlock1) [reply src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx:60]: yep

**Comment by adamwhitlock1** [comment]
> ran e2e tests locally and got one error, specifically for the following file:
> 
> **src/applications/personalization/profile/tests/e2e/profile-a11y.cypress.spec.js**
> 
> Profile -- should pass an aXe scan and manage focus on all pages at desktop size (failed) Profile -- should pass an aXe scan and manage focus on all pages at mobile phone size (failed)
> 
> everything else looks good though 👍

@allisonlu I'm having trouble reproducing a failure on that a11y test. You didn't happen to be running the mock server while running the cypress tests or anything like that? I tested it and ran it in a loop 25 times like a stress test would, and all mine passed, so Im not exactly sure.

**Comment by allisonlu** [comment]
> > ran e2e tests locally and got one error, specifically for the following file:
> > **src/applications/personalization/profile/tests/e2e/profile-a11y.cypress.spec.js**
> > Profile -- should pass an aXe scan and manage focus on all pages at desktop size (failed) Profile -- should pass an aXe scan and manage focus on all pages at mobile phone size (failed)
> > everything else looks good though 👍
> 
> @allisonlu I'm having trouble reproducing a failure on that a11y test. You didn't happen to be running the mock server while running the cypress tests or anything like that? I tested it and ran it in a loop 25 times like a stress test would, and all mine passed, so Im not exactly sure.

ah you're absolutely right... I _did_ have mock server running 🤦🏻‍♀️ that was completely my bad. sorry about the headache.

**Review by allisonlu — APPROVED** [review]

**Pull request merged**